src/picongpu/include/particles/ionization/byField/ADK/AlgorithmADK.hpp

```diff
@@ -82,8 +85,21 @@ namespace ionization
                 float_X dBase = float_X(4.0) * util::cube(protonNumber) / (eInAU * util::quad(nEff)) ;
                 float_X dFromADK = math::pow(dBase,nEff);
 
+                float_X polarizationFactor;
+                /* set up the polarization factor */
+                if(T_linPol)
```

Comment:
The value is not `const` so I cannot just calculate it during compile time (it depends on the ionization potential). Nevertheless, it is known if the bool is `true` or `false` during compile time. Therefore the compiler should optimize the code in such a way that the unnecessary code vanishes.